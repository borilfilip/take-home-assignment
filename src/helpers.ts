export const apiUrl = 'http://localhost:8000';

interface Address {
  city: string;
  street: string;
  number: number;
}

export interface BookingObject {
  id?: string;
  start: Address;
  destination: Address;
  when: string;
  people: number;
}

export interface HTTPValidationError {
  detail: { loc: string; msg: string; type: string }[];
}
