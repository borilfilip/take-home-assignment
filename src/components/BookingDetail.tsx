import { BookingObject } from '../helpers';

interface BookingDetailProps {
  booking?: BookingObject;
}

function BookingDetail({ booking }: BookingDetailProps) {
  return (
    (booking && (
      <>
        <div className="mb-3">
          <strong>Start</strong>
          <br />
          {booking.start.street} {booking.start.number},<br />
          {booking.start.city}
        </div>
        <div className="mb-3">
          <strong>Destination</strong>
          <br />
          {booking.destination.street} {booking.destination.number},<br />
          {booking.destination.city}
        </div>
        <div className="mb-3">
          <strong>Date</strong>
          <br />
          {new Date(booking.when).toLocaleString()}
        </div>
        <div className="mb-3">
          <strong>Number of people</strong>
          <br />
          {booking.people}
        </div>
      </>
    )) ||
    null
  );
}

export default BookingDetail;
