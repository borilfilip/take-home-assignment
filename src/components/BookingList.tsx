import { BookingObject } from '../helpers';
import { Table } from 'react-bootstrap';
import { Link } from 'react-router-dom';

interface BookingListProps {
  bookings?: BookingObject[];
}

function BookingList({ bookings }: BookingListProps) {
  return (
    (bookings?.length && (
      <Table striped hover>
        <thead>
          <tr>
            <th colSpan={3}>Start</th>
            <th colSpan={3}>Destination</th>
            <th>When</th>
            <th>People</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          {bookings.map((booking) => (
            <tr key={booking.id}>
              <td>{booking.start.city}</td>
              <td>{booking.start.street}</td>
              <td>{booking.start.number}</td>
              <td>{booking.destination.city}</td>
              <td>{booking.destination.street}</td>
              <td>{booking.destination.number}</td>
              <td>{new Date(booking.when).toLocaleString()}</td>
              <td>{booking.people}</td>
              <td>
                <Link to={'/detail/' + booking.id} className="btn btn-sm btn-primary">
                  Detail
                </Link>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    )) ||
    null
  );
}

export default BookingList;
