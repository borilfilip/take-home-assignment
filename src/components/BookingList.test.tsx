import renderer from 'react-test-renderer';
import { BookingObject } from '../helpers';
import BookingList from './BookingList';
import { mockBooking } from './BookingDetail.test';
import { BrowserRouter } from 'react-router-dom';

const mockBooking2: BookingObject = {
  id: '2',
  start: {
    city: 'Melbourne',
    street: 'City Rd.',
    number: 183
  },
  destination: {
    city: 'Melbourne',
    street: 'The Parade',
    number: 103
  },
  when: '2022-12-08T08:15:37.164Z',
  people: 2
};

const mockBookings = [mockBooking, mockBooking2];

it('renders booking list', () => {
  const component = renderer.create(
    <BrowserRouter>
      <BookingList bookings={mockBookings} />
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
