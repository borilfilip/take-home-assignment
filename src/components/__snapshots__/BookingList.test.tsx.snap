// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders booking list 1`] = `
<table
  className="table table-striped table-hover"
>
  <thead>
    <tr>
      <th
        colSpan={3}
      >
        Start
      </th>
      <th
        colSpan={3}
      >
        Destination
      </th>
      <th>
        When
      </th>
      <th>
        People
      </th>
      <th>
        Action
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Warszaw
      </td>
      <td>
        Marszałkowska
      </td>
      <td>
        1
      </td>
      <td>
        Praha
      </td>
      <td>
        Staroměstské nám.
      </td>
      <td>
        4
      </td>
      <td>
        12/8/2022, 7:15:37 PM
      </td>
      <td>
        2
      </td>
      <td>
        <a
          className="btn btn-sm btn-primary"
          href="/detail/1"
          onClick={[Function]}
        >
          Detail
        </a>
      </td>
    </tr>
    <tr>
      <td>
        Melbourne
      </td>
      <td>
        City Rd.
      </td>
      <td>
        183
      </td>
      <td>
        Melbourne
      </td>
      <td>
        The Parade
      </td>
      <td>
        103
      </td>
      <td>
        12/8/2022, 7:15:37 PM
      </td>
      <td>
        2
      </td>
      <td>
        <a
          className="btn btn-sm btn-primary"
          href="/detail/2"
          onClick={[Function]}
        >
          Detail
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
